.rustfmt.toml

```diff
@@ -0,0 +1,27 @@
+style_edition = "2024"
```

Comment:
AFAICT most of the changes in this PR are formatting changes. Could we do those separately by setting `style_edition = "2021"` here?

Reply:
Done.